### PR TITLE
Reader Feed Search Results: error handling

### PR DESCRIPTION
**Pull request by samouri**
Related: #13787

screenshot-2017-05-10-14 26 20

/cc @samouri

### CODE REVIEW

File: client/reader/following-manage/feed-search-results.jsx [hunk 1]
```diff
@@ -22,14 +23,24 @@ const FollowingManageSearchFeedsResults = ( {
 	fetchNextPage,
 	forceRefresh,
 	searchResultsCount,
+	query,
 } ) => {
+	const isError = !! ( query && searchResults.length === 0 );
+	const classNames = classnames( 'following-manage__search-results', {
+		'is-error': !! isError,
```
Comment (jancavan) [inline client/reader/following-manage/feed-search-results.jsx]: If this is intended for `no-results` only, maybe class name should be `is-empty` instead?

**Comment by jancavan** [comment]
@samouri Sometimes it says "No results":
screenshot 2017-05-10 14 00 40

But other times other queries throw an error:
screenshot 2017-05-10 14 02 29

File: client/reader/following-manage/feed-search-results.jsx [hunk 2]
```diff
@@ -22,14 +23,24 @@ const FollowingManageSearchFeedsResults = ( {
...
 	if ( ! searchResults ) {
 		return null; // todo: add placeholder
 	} else if ( searchResults.length === 0 ) {
 		return (
-			<p>
-				{ translate( 'There were no site results for your query.' ) }
-			</p>
+			<div className={ classNames }>
+				<p>
+					{ translate( 'Sorry, no sites match %s.', {
```
Comment (jancavan) [inline client/reader/following-manage/feed-search-results.jsx]: How do we italicize the query?

**Comment by jancavan** [comment]
@samouri I've updated the description in this PR with a mockup on how no-results should look in Site Search and Following filter.

File: client/reader/following-manage/feed-search-results.jsx (hunk 2 above)
Comment (bluefuton) [inline client/reader/following-manage/feed-search-results.jsx]: You can use the `components` cleverness in `translate`, like this:

```{ translate(
	'Enter your {{strong}}WordPress.com{{/strong}} username or email address',
	{ components: { strong: <strong /> } }
) }

**Comment by samouri** [comment]
@jancavan, could you give this some 👀 ?

**Comment by jancavan** [comment]
@samouri Made minor changes in: https://github.com/Automattic/wp-calypso/pull/13834/commits/27bb7024b0c7dc2a94fd22eee2aa75751932a7aa. I think this is good to go after a rebase.

**Comment by samouri** [comment]
rebased

**Comment by samouri** [comment]
@jancavan: lets rescope this PR to only deal with introducing the new empty result displays.

The actual api errors and error notices we should handle separately

**Comment by bluefuton** [comment]
Feels a touch close to the input for me, but works well 👍 

screen shot 2017-05-13 at 10 00 44

File: client/reader/following-manage/style.scss [hunk 3]
```diff
@@ -202,6 +202,17 @@
 	}
 }
 
+.following-manage__search-results.is-empty {
+	color: darken( $gray, 10% );
+}
+
+.following-manage__subscriptions-list.is-empty {
+	color: darken( $gray, 10% );
+	p {
```
Comment (bluefuton) [inline client/reader/following-manage/style.scss]: We should target this `p` directly with a class name rather than styling the element.

**Comment by bluefuton** [comment]
Now I'm confused :) I fixed the class name nit above and the message is now centred. Is that the correct position @jancavan?

screen shot 2017-05-13 at 11 47 29

Looks better I reckon 👍

File: client/reader/following-manage/style.scss (hunk 3 above)
Comment (bluefuton) [inline client/reader/following-manage/style.scss]: Fixed 👍

File: client/reader/following-manage/feed-search-results.jsx [hunk 4]
```diff
@@ -22,14 +23,25 @@ const FollowingManageSearchFeedsResults = ( {
 	fetchNextPage,
 	forceRefresh,
 	searchResultsCount,
+	query,
 } ) => {
+	const isEmpty = !! ( query && searchResults && searchResults.length === 0 );
+	const classNames = classnames( 'following-manage__search-results', {
+		'is-empty': !! isEmpty,
```
Comment (blowery) [inline client/reader/following-manage/feed-search-results.jsx]: this doesn't need the `!!`

File: client/reader/following-manage/feed-search-results.jsx [hunk 5]
```diff
@@ -22,14 +23,25 @@ const FollowingManageSearchFeedsResults = ( {
...
-	} else if ( searchResults.length === 0 ) {
+	} else if ( isEmpty ) {
 		return (
-			<p>
-				{ translate( 'There were no site results for your query.' ) }
-			</p>
+			<div className={ classNames }>
+				<p className="following-manage__search-results-empty-message">
+					{ translate( 'Sorry, no sites match {{italic}}%s.{{/italic}}', {
+						components: { italic: <i /> },
```
Comment (blowery) [inline client/reader/following-manage/feed-search-results.jsx]: this should probably use `<em />` instead of `<i />`

Comment (blowery) [inline client/reader/following-manage/feed-search-results.jsx]: actually, nevermind. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/em#<i>_vs._<em>

Comment (bluefuton) [inline client/reader/following-manage/feed-search-results.jsx]: I had exactly the same feedback, but discovered `<i />` now has semantic value in HTML5 and is probably the most appropriate element in that case.

File: client/reader/following-manage/style.scss (hunk 3 above)
Comment (jancavan) [inline client/reader/following-manage/style.scss]: @bluefuton I removed the <p> tag here since this isn't really a paragraph.

**Comment by jancavan** [comment]
> Is that the correct position

The message for empty Search results should be left-aligned. Fixed in: https://github.com/Automattic/wp-calypso/pull/13834/commits/76207e0af8870dba3d6ae83bd963fcf84e6eff0d

**Comment by samouri** [comment]
@jancavan, all the code LGTM. I'll give it another round of testing and 🚢  if good

**Pull request merged**